File: 2000 98 & ME/QuickStart/howto/samples/Services/Process/processdemo/processdemo.cs
using System;
using System.Diagnostics;
using System.Threading;

public class ProcessDemo
{
    public static void Main(String[] args) 
    {
        string appName = Environment.GetCommandLineArgs()[0];
        
        if(args.Length != 1)
        {
            Console.WriteLine("Usage: " + appName +" <executable>");
            return;
        }
        string executableFilename = args[0];
        
        Process process = new Process();
        process.StartInfo.FileName = executableFilename;
        process.Start();
        
        process.WaitForInputIdle();
        
        Thread.Sleep(1000);
        if(!process.CloseMainWindow())
        {
            process.Kill();
        }
    }
}

File: 2000 98 & ME/QuickStart/howto/samples/Services/Process/processdemo/vb/ProcessDemo.vb
Imports System
Imports System.Diagnostics
Imports System.Threading

Namespace ProcessDemo

    Module ProcessDemo

        Public Sub Main()
            Dim args As String()
            Dim appName As String
            args = Environment.GetCommandLineArgs()
            appName = args(0)
        
            If (args.Length <> 2) Then
                Console.WriteLine("Usage: " + appName + " <executable>")
                Exit Sub
            End If

            Dim executableFilename As String
            executableFilename = args(1)

            Dim process As Process
            process = New Process
            process.StartInfo.FileName = executableFilename
            process.Start()
        
            process.WaitForInputIdle()
        
            Thread.Sleep(1000)
            If (Not process.CloseMainWindow()) Then
                process.Kill()
            End If

        End Sub

    End Module

End Namespace
